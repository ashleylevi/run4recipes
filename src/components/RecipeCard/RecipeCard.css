.recipe-card {
  border: .5px solid black;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 400px;
  width: 250px;
  margin: 5px 5px 100px 5px;
  transition: 0.3s;
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  position: relative;
  /* filter: blur(1px) */
}

a {
  text-decoration: none;
  color: black;
}

.favorites-box {
  position: absolute;
  top: 350px;
  left: 20px;
  width: 200px;
  height: 100px;
  background: #FFF;
  border: 4px solid #32636E;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.recipe-name {
  font-family: Raleway;
  font-style: normal;
  font-weight: 600;
  line-height: normal;
  font-size: 15px;
  color: #3D323E;
  text-align: center;
  margin: 20px 0 10px 0;
}

.fa-heart {
  margin: 0 auto;
  font-size: 30px;
  margin: 5px 0 10px 0;
}

.fa-heart:hover {
  cursor: pointer;
  color: #D5AD47;
}

.fas {
  color: #D5AD47;
}