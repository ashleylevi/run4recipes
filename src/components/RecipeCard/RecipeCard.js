import React from 'react';
import './RecipeCard.css';

export const RecipeCard = ({recipe}) => {
  console.log(recipe)
  return (
    <div className="recipe-card" style={{ backgroundImage: `url(${recipe.image})` }}>
    <p>{recipe.name}</p>

    </div>
  )
}