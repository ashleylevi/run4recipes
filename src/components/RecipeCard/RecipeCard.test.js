import React from 'react';
import { shallow } from 'enzyme';
import { RecipeCard } from './RecipeCard';

describe('RecipeCard', () => {
  let wrapper;
  let mockRecipe;

  beforeEach(() => {
    mockRecipe  = {
      label: "Spaghetti and meatballs",
      image: "https://image.png",
      ingredientLines: [],
      category: 'bread'
    }
    wrapper = shallow(<RecipeCard recipe={mockRecipe} />);
  });

  it('Should render like snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});