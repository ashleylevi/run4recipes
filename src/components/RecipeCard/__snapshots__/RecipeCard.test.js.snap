// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeCard Should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RecipeCard
    recipe={
      Object {
        "category": "spaghetti",
        "image": "https://image.png",
        "ingredients": Array [
          "spaghetti",
          "meatballs",
        ],
        "name": "Spaghetti and meatballs",
      }
    }
    updateFavorites={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="gradient"
        />,
        <div
          className="favorites-box"
        >
          <p
            className="recipe-name"
          >
            Spaghetti and meatballs
          </p>
          <i
            className="far fa-heart"
            onClick={[Function]}
          />
        </div>,
      ],
      "className": "recipe-card",
      "style": Object {
        "backgroundImage": "url(https://image.png)",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "gradient",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="recipe-name"
            >
              Spaghetti and meatballs
            </p>,
            <i
              className="far fa-heart"
              onClick={[Function]}
            />,
          ],
          "className": "favorites-box",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Spaghetti and meatballs",
              "className": "recipe-name",
            },
            "ref": null,
            "rendered": "Spaghetti and meatballs",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "far fa-heart",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="gradient"
          />,
          <div
            className="favorites-box"
          >
            <p
              className="recipe-name"
            >
              Spaghetti and meatballs
            </p>
            <i
              className="far fa-heart"
              onClick={[Function]}
            />
          </div>,
        ],
        "className": "recipe-card",
        "style": Object {
          "backgroundImage": "url(https://image.png)",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "gradient",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="recipe-name"
              >
                Spaghetti and meatballs
              </p>,
              <i
                className="far fa-heart"
                onClick={[Function]}
              />,
            ],
            "className": "favorites-box",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Spaghetti and meatballs",
                "className": "recipe-name",
              },
              "ref": null,
              "rendered": "Spaghetti and meatballs",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "far fa-heart",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
